docs/CustomTranslator.md

```diff
@@ -111,4 +111,55 @@ class FakeTranslator {
 FakeTranslator;
 ```
 
+Example of custom translator using [Lingva (Alternative front-end for Google Translate ) API](https://github.com/thedaviddelta/lingva-translate):
+
+```js
+class LingvaTranslator {
+    // URL list of instance of LingvaTranslate, add more to use fastest one using Promise.race()
+    apiList = ["https://lingva.ml"]
+
+    translate = (text, from, to) => {
+        let promiseList = this.apiList.map(api => fetch(`${api}/api/v1/${from}/${to}/${text}`, {
+            credentials: "omit",
+            headers: {
+                "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:99.0) Gecko/20100101 Firefox/99.0",
+                Accept: "*/*",
+                "Accept-Language": "en-US,en;q=0.5",
+                "Sec-Fetch-Dest": "empty",
+                "Sec-Fetch-Mode": "cors",
+                "Sec-Fetch-Site": "same-origin"
+            },
+            method: "GET",
+            mode: "cors",
+        }))
+        return Promise.race(promiseList)
+            .then((r) => r.json())
+            .then(({ translation }) => translation);
+    };
+
+    translateBatch = (texts, from, to) =>
+        Promise.all(texts.map((text) => this.translate(text, from, to)));
+
+    getLengthLimit = () => 4000;
+    getRequestsTimeout = () => 1000;
+    checkLimitExceeding = (text) => {
+        const textLength = !Array.isArray(text) ? text.length : text.reduce((len, text) => len + text.length, 0);
+
+        return textLength - this.getLengthLimit();
+    }
+
+    static isSupportedAutoFrom = () => true;
+    static getSupportedLanguages = () => [
+        "en", "ar", "az", "zh", "cs",
+        "nl", "eo", "fi", "fr", "de",
+        "el", "hi", "hu", "id", "ga",
+        "it", "ja", "ko", "fa", "pl",
+        "pt", "ru", "sk", "es", "sv",
+        "tr", "uk", "vi"
+    ];
```

Comment:
Is this actual list of languages for any instance? Where we can found this information?

Reply:
not sure either, but since lingva is just a proxy to google translate, I think the languages can be supported

Reply:
I've read the doc https://github.com/thedaviddelta/lingva-translate#rest-api-v1
As i see you can use URL https://lingva.ml/api/v1/languages to get languages list.
Just format it to array with language codes and let's go to merge

Reply:
you mean all, I see there are 133 codes in total (except for 'auto')

Reply:
Yes, it is correct

Reply:
I suggest snippet to parse languages from page https://lingva.ml

```js
await fetch('/api/v1/languages').then(r => r.json()).then(langs => Object.entries(langs.languages).slice(1).map(([_, lang]) => lang.code))
```

Reply:
yep, done

Reply:
Let's remove `auto` and split this line to a few lines. It need to users can see whole code with no scrolling